### PR TITLE
Added log4j2 dependency for sample-auth.war

### PR DESCRIPTION
**Pull request by hwupathum**
fixes https://github.com/wso2/product-is/issues/14864

### CODE REVIEW

**Review by nilminiwso2 — APPROVED** [review]

File: authenticators/components/org.wso2.carbon.identity.sample.extension.auth.endpoint/pom.xml [hunk 1]
```diff
@@ -69,6 +69,11 @@
             <artifactId>org.wso2.carbon.identity.application.authentication.framework</artifactId>
             <scope>provided</scope>
         </dependency>
+        <dependency>
+            <groupId>org.apache.logging.log4j</groupId>
+            <artifactId>log4j-core</artifactId>
+            <version>2.18.0</version>
```
Comment (piraveena) [inline authenticators/components/org.wso2.carbon.identity.sample.extension.auth.endpoint/pom.xml:75]: move the version to the parent pom file

File: authenticators/components/org.wso2.carbon.identity.sample.extension.auth.endpoint/src/main/webapp/hwk.jsp [hunk 2]
```diff
@@ -20,10 +20,11 @@
 <%@ page import="java.net.URLDecoder" %>
 <%@ page import="java.nio.charset.StandardCharsets" %>
 <%@ page import="org.wso2.carbon.identity.application.authentication.framework.util.FrameworkUtils" %>
-<%@page import="org.apache.log4j.Logger"%>
+<%@ page import="org.apache.logging.log4j.Logger"%>
+<%@ page import="org.apache.logging.log4j.LogManager"%>
```
Comment (chamathns) [inline authenticators/components/org.wso2.carbon.identity.sample.extension.auth.endpoint/src/main/webapp/hwk.jsp:24]: shall we use apache.commons.logging or slf4j here?

**Comment by sadilchamishka** [comment]
It would be a best practice to use logging facade like Slf4j or apache commons logging library instead of concrete logging implementations. Please refer this https://github.com/wso2/carbon-identity-framework/pull/3982/files

**Review by sadilchamishka — APPROVED** [review]

File: authenticators/components/org.wso2.carbon.identity.sample.extension.auth.endpoint/pom.xml [hunk 3]
```diff
@@ -69,6 +69,11 @@
             <artifactId>org.wso2.carbon.identity.application.authentication.framework</artifactId>
             <scope>provided</scope>
         </dependency>
+        <dependency>
+            <groupId>org.slf4j</groupId>
+            <artifactId>slf4j-log4j12</artifactId>
+            <version>${slf4j.version}</version>
+        </dependency>
```
Comment (sadilchamishka) [inline authenticators/components/org.wso2.carbon.identity.sample.extension.auth.endpoint/pom.xml:76]:
```suggestion
        <dependency>
                <groupId>org.slf4j</groupId>
                <artifactId>slf4j-api</artifactId>
                <version>${org.slf4j.verison}</version>
</dependency>

**Review by piraveena — APPROVED** [review]

**Pull request merged**